conv/convert_types_test.go

```diff
@@ -0,0 +1,203 @@
+	})
+
+	t.Run("with PointerSlice on []string", func(t *testing.T) {
+		testCasesStringSlice := [][]string{
+			{"a", "b", "c", "d", "e"},
+			{"a", "b", "", "", "e"},
+		}
+
+		for idx, in := range testCasesStringSlice {
+			if in == nil {
+				continue
+			}
+			out := PointerSlice(in)
+			assertValues(t, in, out, wantsPointer, idx)
+
+			out2 := ValueSlice(out)
+			assertValues(t, in, out2, wantsValue, idx)
+		}
+	})
+
+	t.Run("with ValueSlice on []string", func(t *testing.T) {
+		testCasesStringValueSlice := [][]*string{
+			{Pointer("a"), Pointer("b"), nil, Pointer("c")},
+		}
+
+		for idx, in := range testCasesStringValueSlice {
+			if in == nil {
+				continue
+			}
+			out := ValueSlice(in)
+			assertValues(t, in, out, wantsValue, idx)
+
+			out2 := PointerSlice(out)
+			assertValues(t, in, out2, wantsPointer, idx)
+		}
+	})
+}
+
+func TestMap(t *testing.T) {
+	t.Run("with nulls", func(t *testing.T) {
+		require.Empty(t, ValueMap[string, uint16](nil))
+
+		require.Len(t, ValueMap(map[string]*int{"a": Pointer(1), "b": nil, "c": Pointer(2)}), 2)
+	})
+
+	t.Run("with PointerMap on map[string]string", func(t *testing.T) {
+		testCasesStringMap := []map[string]string{
+			{"a": "1", "b": "2", "c": "3"},
+		}
+
+		for idx, in := range testCasesStringMap {
+			if in == nil {
+				continue
+			}
+			out := PointerMap(in)
+			assertValues(t, in, out, wantsPointer, idx)
+
+			out2 := ValueMap(out)
+			assertValues(t, in, out2, wantsValue, idx)
+		}
+	})
+
+	t.Run("with ValueMap on map[string]bool", func(t *testing.T) {
+		testCasesBoolMap := []map[string]bool{
+			{"a": true, "b": false, "c": true},
+		}
+
+		for idx, in := range testCasesBoolMap {
+			if in == nil {
+				continue
+			}
+			out := PointerMap(in)
+			assertValues(t, in, out, wantsPointer, idx)
+
+			out2 := ValueMap(out)
+			assertValues(t, in, out2, wantsValue, idx)
+		}
+	})
+}
+
+func TestPointer(t *testing.T) {
+	t.Run("with Pointer on string", func(t *testing.T) {
+		testCasesString := []string{"a", "b", "c", "d", "e", ""}
+
+		for idx, in := range testCasesString {
+			out := Pointer(in)
+			assertValues(t, in, out, wantsPointer, idx)
+
+			out2 := Value(out)
+			assertValues(t, in, out2, wantsValue, idx)
+		}
+		assert.Zerof(t, Value[string](nil), "expected conversion from nil to return zero value")
+	})
+
+	t.Run("with Value on bool", func(t *testing.T) {
+		testCasesBool := []bool{true, false}
+
+		for idx, in := range testCasesBool {
+			out := Pointer(in)
+			assertValues(t, in, out, wantsPointer, idx)
+
+			out2 := Value(out)
+			assertValues(t, in, out2, wantsValue, idx)
+		}
+		assert.Zerof(t, Value[bool](nil), "expected conversion from nil to return zero value")
+	})
+}
+
+func assertSingleValue(t *testing.T, inElem, elem reflect.Value, expectPointer bool, idx int) {
+	require.Equalf(t,
+		expectPointer, (elem.Kind() == reflect.Ptr),
+		"unexpected expectPointer=%t value type", expectPointer,
```

Comment:
The type you received should also be mentioned

```suggestion
		"unexpected expectPointer=%t value type, got %T", expectPointer, elem.Kind(),
```

Reply:
yes. Same response. I've overlooked these existing utilities. Will fix it